test/cbmc/proofs/stringBuilderUInt32Hex/stringBuilderUInt32Hex_harness.c

```diff
@@ -0,0 +1,47 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/*
+ * Insert copyright notice
+ */
+
+/**
+ * @file stringBuilderUInt32Hex_harness.c
+ * @brief Implements the proof harness for stringBuilderUInt32Hex function.
+ */
+
+/*
+ * Insert project header files that
+ *   - include the declaration of the function
+ *   - include the types needed to declare function arguments
+ */
+#include "ota_mqtt_private.h"
+
+#define U32_MAX_LEN    10U
+
+/* Declaration of the mangled name function generated by CBMC for static functions. */
+size_t __CPROVER_file_local_ota_mqtt_c_stringBuilderUInt32Hex( char * pBuffer,
+                                                               size_t bufferSizeBytes,
+                                                               uint32_t value );
+
+void stringBuilderUInt32Hex_harness()
+{
+    char * pBuffer;
+    size_t bufferSizebytes;
+    uint32_t value;
+
+    /* The bufferSizebytes is the size of the pBuffer. The pBuffer is statically initialized with
+     *  a size of U32_MAX_LEN + 1 in all the functions which call stringBuilderUInt32Decimal. Hence,
+     *  the size can never be below that. */
+    __CPROVER_assume( bufferSizebytes > U32_MAX_LEN );
```

Comment:
This assumption doesn't quite make sense to me.
1. Do we need an assumption for this?
2. If we do need an assumption, why make it go from the expected size all the way up to the maximum possible value?

I see your comment that it can never be below that size, but does that not also mean that it can't be above that size?

Reply:
This assumption is required because the size of the pBuffer should always be above the U32_MAX_LEN. If the size of the goes below the size U32_MAX_LEN then it hits the assert statement in the source code. The assumption does not have a max constraint since in any case if the code is modified with the size of pBuffer greater than U32_MAX_LEN the CBMC proofs should not fail.

Reply:
I see, thanks for clarifying.